porch/pkg/cache/repository.go

```diff
@@ -258,63 +264,55 @@ func (r *cachedRepository) update(ctx context.Context, updated repository.Packag
 	// TODO: Just updated package?
 	identifyLatestRevisions(r.cachedPackageRevisions)
 
-	// TODO: Update the latest revisions for the r.cachedPackages
 	return cached, nil
 }
 
 func (r *cachedRepository) DeletePackageRevision(ctx context.Context, old repository.PackageRevision) error {
+	// get the reconcile lock first, before touching the underlying repo
+	r.reconcileMutex.Lock()
+	defer r.reconcileMutex.Unlock()
+
 	// Unwrap
 	unwrapped := old.(*cachedPackageRevision).PackageRevision
 	if err := r.repo.DeletePackageRevision(ctx, unwrapped); err != nil {
 		return err
 	}
 
 	r.mutex.Lock()
-	if r.cachedPackages != nil {
+	if r.cachedPackageRevisions != nil {
 		k := old.Key()
-		// previous := r.cachedPackages[k]
 		delete(r.cachedPackageRevisions, k)
 
 		// Recompute latest package revisions.
 		// TODO: Only for affected object / key?
 		identifyLatestRevisions(r.cachedPackageRevisions)
 	}
-
 	r.mutex.Unlock()
 
 	return nil
 }
 
 func (r *cachedRepository) ListPackages(ctx context.Context, filter repository.ListPackageFilter) ([]repository.Package, error) {
-	packages, err := r.getPackages(ctx, filter, false)
-	if err != nil {
-		return nil, err
-	}
-
-	return packages, nil
+	return nil, fmt.Errorf("not implemented")
```

Comment:
We can probably delete the Package functions. This was added as part of a plan to support Package as a separate resource, but it was never completed.

Reply:
Yes, that's my plan eventually but not in this PR.